### PR TITLE
Fix console paging and hardware bank synchronization (#144)

### LINKED ISSUE #144: Console bank switching desyncs MIDI bank bindings and the MCU surface
src/ui/ConsoleView.cpp:153 and :161 (release/0.12)

setBank stores the width-derived screen bank index into Session::activeBank, but the audio thread resolves it as bank * Session::kBankSize (8) at AudioEngine.cpp:3069. The screen stride is channelsThatFit(), clamped to [6, 24], which is 8 only by coincidence. At fit 6 the console shows tracks 6-11 on bank 1 while bank-relative MIDI bindings drive tracks 8-15; screen bank 3 gives base 24 and every TrackFaderBank/PanBank/MuteBank binding no-ops against the kNumTracks guard.

Same index goes into session.mcu.bank (:161), which the MCU code resolves as bank * kStripsPerBank (8) and whose own Bank Right clamps at kNumBanks-1 (2). numBanks() can be 4, so picking screen bank 3 writes mcu.bank 3, every MCU strip resolves to tracks 24-31, and the surface goes fully dead with no way back from the surface itself.

Fix: keep the screen scroll position separate from the 3x8 hardware bank model, or translate before storing.

### PR DESCRIPTION
**Pull request by marc-korte**
## What

- Keep width-dependent console pages separate from fixed 3x8 hardware banks.
- Publish the hardware bank containing the first visible track and mirror MCU bank changes back to the matching screen page.
- Preserve surface-derived banks across resize while recalculating screen-derived banks.
- Add focused mapping and synchronization regression coverage.

## Why

Console page strides range from 6 to 24 channels, while bindings and MCU banks always use groups of 8. Reusing a screen page index as a hardware bank desynchronized visible tracks from controls and could publish an out-of-range MCU bank.

## Validation

- DuskStudio app build
- 537/537 tests passed
- JUCE gate: 179 files / 9257 uses

Fixes #144

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved synchronization between visible screen pages and hardware bank selections.
  * Preserved the correct active bank when resizing and prevented unnecessary bank updates.
  * Added boundary handling for invalid or out-of-range bank and page values.

* **Tests**
  * Added coverage for bank mapping across supported layouts, shared banks, identity mappings, and edge cases.
  * Added regression checks to ensure resolved banks remain within valid limits.

### CODE REVIEW

**Comment by marc-korte** [comment]
@coderabbitai review

**Comment by marc-korte** [comment]
@coderabbitai review

File: src/ui/BankMapping.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "../session/Session.h"
```
[the remark this answers is not in the document]
Comment (marc-korte) [reply src/ui/BankMapping.h]: Fixed in 899bdf3. The shared track and bank geometry now lives in dependency-free SessionLayout.h; Session preserves its existing public aliases, and BankMapping.h no longer includes Session.h or pulls in JUCE transitively.

File: src/ui/ConsoleView.cpp [hunk 2]
```diff
@@ -201,24 +230,24 @@ void ConsoleView::resized()
...
+
+    // Both the page index and the stride under it are width-derived, so a
+    // resize moves which hardware bank the page sits in - re-derive, or
+    // bank-relative bindings keep driving whichever strips the OLD width put
+    // under the page. mcu.bank is deliberately untouched: the surface's own
+    // position is width-independent, and with a single page the screen has no
+    // bank opinion to impose in the first place.
+    if (pages > 1 && activeBankFollowsScreen)
+        sessionRef.activeBank.store (hardwareBankForScreenBank (currentBank, stride),
+                                     std::memory_order_relaxed);
```
[the remark this answers is not in the document]
Comment (marc-korte) [reply src/ui/ConsoleView.cpp]: Fixed in 899bdf3. Screen-owned resize remaps now update lastKnownMcuBank, activeBank, and mcu.bank together. The new regression covers page 3 at stride 6 widening to stride 12, clamping to page 1 and bank 1 for all three published values.
